telos-backend/src/routes/api/journal.js

```diff
@@ -1,14 +1,119 @@
 const express = require('express');
+const CalendarEvent = require('../../models/calendar_event');
+const Habit = require('../../models/habit');
+const Text = require('../../models/text');
+const TodoTask = require('../../models/todo_task');
+const Widget = require('../../models/widget');
 
 const router = express.Router();
 
+// Helper methods ---------------------------------
+// Return a string representation of current day. 
+// See daysOfWeekenum in src/models/habit.js
+function getStringDayOfWeek(dayNum) {
+  let dayString;
+  switch (dayNum) {
+    case 0:
+      dayString = 'sun';
+      break;
+    case 1:
+      dayString = 'mon';
+      break;
+    case 2:
+      dayString = 'tue';
+      break;
+    case 3:
+      dayString = 'wed';
+      break;
+    case 4:
+      dayString = 'thu';
+      break;
+    case 5:
+      dayString = 'fri';
+      break;
+    case 6:
+      dayString = 'sat';
+      break;
+    default:
+      dayString = 'err';
+  }
+  return dayString;
+}
+
+async function getTodosForDay(date){
+  // Todos should return either what is made today, or all overdue tasks.
```

Comment:
I think the tasks that should be returned are ones that are either:

- made today
- due today
- overdue (due before today but is not marked as completed)

So I think the logic for the query should be a little different, something like:
$or: [
    { createdDate: { $gte: date + 'T00:00:00', $lte: date + 'T23:59:59' } }, 
    { dueDate: { $gte: date + 'T00:00:00', $lte: date + 'T23:59:59' } },
    { dueDate: { $lt: date + 'T00:00:00' }, completed: false }
]

Reply:
You're right, not currently getting overdue ones, and would return stuff not due today - will change this query.